### PR TITLE
Refactor user service

### PR DESCRIPTION
**Pull request by tyanhan**
- Refactored status codes to userController
- Added checks for password security

### CODE REVIEW

**Comment by tyanhan** [comment]
For the sign up function, is it correct to check whether a username already exists in the controller layer? It sounds like business logic but I am not sure how to shift it to the domain layer and still return the corresponding status code.

File: user-service/domain/userDomain.js [hunk 1]
```diff
@@ -1,32 +1,56 @@
...
+			lower = /[a-z]/.test(string),
+			specialCharacter = /[`!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/.test(string);
+		return upper && lower && specialCharacter && password.length >= 8;
+	}
+
+	if (!isPasswordValid(password)) {
+		return false;
+	}
+
+	console.log("Password meets requirements");
```
Comment (kevin9foong) [inline user-service/domain/userDomain.js:24]: Should set to debug

**Review by kevin9foong — APPROVED** [review]
Just make the minor changes then LGTM

**Comment by kevin9foong** [comment]
yeah, its ok to put in controller layer, theres no way to avoid it anyways and its simple to understand. the idea is just to abstract implementation detail eg, getting record from database in the domain layer

**Comment by tyanhan** [comment]
Since we are changing to 3rd party services for user service, I think we can close this PR?

**Pull request closed without merging**